imap_processing/ena_maps/utils/spatial_utils.py

```diff
@@ -0,0 +1,270 @@
+"""IMAP utils for spatial binning and az/el grid creation."""
+
+from __future__ import annotations
+
+import typing
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+def build_spatial_bins(
+    az_spacing: float = 0.5,
+    el_spacing: float = 0.5,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Build spatial bin boundaries for azimuth and elevation (input/output angles in deg).
+
+    Parameters
+    ----------
+    az_spacing : float, optional
+        The azimuth bin spacing in degrees (default is 0.5 degrees).
+    el_spacing : float, optional
+        The elevation bin spacing in degrees (default is 0.5 degrees).
+
+    Returns
+    -------
+    az_bin_edges : np.ndarray
+        Array of azimuth bin boundary values in degrees.
+    el_bin_edges : np.ndarray
+        Array of elevation bin boundary values in degrees.
+    az_bin_midpoints : np.ndarray
+        Array of azimuth bin midpoint values in degrees.
+    el_bin_midpoints : np.ndarray
+        Array of elevation bin midpoint values in degrees.
+    """
+    # Azimuth bins from 0 to 360 degrees.
+    az_bin_edges = np.arange(0, 360 + az_spacing, az_spacing)
+    az_bin_midpoints = az_bin_edges[:-1] + az_spacing / 2  # Midpoints between edges
+
+    # Elevation bins from -90 to 90 degrees.
+    el_bin_edges = np.arange(-90, 90 + el_spacing, el_spacing)
+    el_bin_midpoints = el_bin_edges[:-1] + el_spacing / 2  # Midpoints between edges
+
+    return az_bin_edges, el_bin_edges, az_bin_midpoints, el_bin_midpoints
+
+
+def build_solid_angle_map(
+    spacing: float, input_degrees: bool = True, output_degrees: bool = False
+) -> NDArray:
+    """
+    Build a solid angle map for a given spacing in degrees.
+
+    Parameters
+    ----------
+    spacing : float
+        The bin spacing in the specified units.
+    input_degrees : bool, optional
+        If True, the input spacing is in degrees
+        (default is True for radians).
+    output_degrees : bool, optional
+        If True, the output solid angle map is in square degrees
+        (default is False for steradians).
```

Comment:
Do we need the options here? Sometimes it is helpful to limit our set of input parameters and document clearly what you are doing, then force the caller to do the transforms if they want something different.

Your docstring here says: build "solid angle map for a given spacing in degrees", but then you allow `input_degrees` to change that which is a bit confusing. You can change the first space parameter to be spacing_deg if you want to be explicit too.

Reply:
Done! I know I tend to design functions to be too flexible, with too many params, etc.

I chose to use `spacing_deg`, as you suggest, and always output steradians (while not the most self consistent, I think those are the right units, as the maps are built in degree increments, but square degrees are unpleasant).